base/deprecated.jl

```diff
@@ -363,6 +363,9 @@ end
 
 @deprecate ipermutedims(A::AbstractArray,p) permutedims(A, invperm(p))
 
+# PR #25168
+@deprecate ipermute!(a, p::AbstractVector) invpermute!(a, p::AbstractVector)
```

Comment:
I don't think the deprecation target should have the type annotation, since `@deprecate` turns it into function definition which means that on the RHS it's a type assertion.

Reply:
Right, thanks!